tensorflow/compiler/xla/service/backend.cc

```diff
@@ -142,7 +142,9 @@ Backend::Backend(se::Platform* platform, Compiler* compiler,
   }
 }
 
-Backend::~Backend() {}
+Backend::~Backend() {
+    CHECK_EQ(memory_allocator_.use_count(), 1);
```

Comment:
Why is this true?  What prevents a compilation from running concurrently with backend destruction?  (Please add a comment.)

Reply:
check is not needed. Memory allocator is stored in shared pointer, so even if Backend is destroyed, compilation will succeed.